src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -414,108 +415,119 @@ bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string
             var labelHistogramName = ctx.AddInitializer(labelHistogramExpanded, new long[] { _featureHistogram[0].Length, _labelHistogram.Length }, "labelHistogramExpanded");
             var learnedAbsentFeatureLogProb = ctx.AddInitializer(_absentFeaturesLogProb, new long[] { _absentFeaturesLogProb.Length, 1 }, "absentFeaturesLogProb");
 
-            var greaterOutput = ctx.AddIntermediateVariable(null, "greaterOutput", true);
+            var greaterOutput = ctx.AddIntermediateVariable(BooleanDataViewType.Instance, "greaterOutput", true);
             var opType = "Greater";
```

Comment:
I don't think type information gets added if the last parameter is true. Can you please confirm this? 
Adding the type and shape information makes it easier to read and debug the model. There are only a few cases I have seen where adding type information causes problems. Let us try to add type information as far as possible (in all the calls below) and skip adding type information only if it is causing problems (and let us document the problem)

Reply:
Also, do these fixes fix the tests in Nimbus?

---
In reply to: [390592978](https://github.com/dotnet/machinelearning/pull/4928#discussion_r390592978) [](ancestors = 390592978)

Reply:
Yeah, the info doesn't get added. I think adding the types would make it easier to read and debug especially for the casting nodes, even if the info doesn't get added to the graph (something similar is done in OneVersusAll transformer). I have removed the true from some variables, but I can't do that for most variables in this trainer, since the tensor shapes change from the original shape.

Reply:
Yes, this fixes Nimbus.

Reply:
If you look at OneVersusTrainer, shape information is added in most cases (except for a few calls in SaveAsOnnxPreProcess. If the tensor shape changes, is it possible to identify the shape and encode it?

---
In reply to: [390622962](https://github.com/dotnet/machinelearning/pull/4928#discussion_r390622962) [](ancestors = 390622962)

Reply:
Yup. Added the shape info.